activation/nipost.go

```diff
@@ -479,3 +495,21 @@ func constructMerkleProof(challenge types.Hash32, members []types.Member) (*type
 		Nodes:     nodesH32,
 	}, nil
 }
+
+func randomDurationInRange(min, max time.Duration) time.Duration {
+	return min + time.Duration(float64((max-min).Nanoseconds())*rand.Float64())
+}
+
+// Time to wait before querying for the proof
+// We add a jitter to avoid all nodes querying for the proof at the same time.
+// There's no need to add jitter if the round has already ended.
+func calcGetProofWaitTime(roundEnd time.Time, cycleGap time.Duration) (waitTime, jitter time.Duration) {
+	waitTime = time.Until(roundEnd)
+	if waitTime > 0 {
```

Comment:
I think this should check for `waitTime > -jitter` so that if the node is a bit late it still jitters correctly.